partiql-catalog/src/call_defs.rs

```diff
@@ -87,3 +87,23 @@ impl Debug for CallSpec {
         write!(f, "CallSpec [{:?}]", &self.input)
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct ScalarFnCallDef {
+    pub names: Vec<&'static str>,
+    pub overloads: ScalarFnCallSpecs,
+}
+
+pub type ScalarFnCallSpecs = Vec<ScalarFnCallSpec>;
+
+#[derive(Clone)]
+pub struct ScalarFnCallSpec {
+    pub input: Vec<CallSpecArg>,
+    pub output: Box<dyn ScalarFnExpr>,
+}
```

Comment:
Does it also need to include  `isNullCall` and `isMissingCall` as `partiql-lang-kotlin` [implements](https://github.com/partiql/partiql-lang-kotlin/blob/da22cf8665e1dcf0b3747452285cca6df73c0e95/partiql-spi/src/main/kotlin/org/partiql/spi/function/Function.kt#L28-L29)?

Reply:
Perhaps in the fullness of time. I think that will become more apparent (and more testable) when the existing builtins are migrated into the catalolg.

Reply:
See #499